### DISCUSSION TITLE
Problem running st-flash

### DISCUSSION THREAD

**Discussion by Hugo-Habicht** [stlink-org/stlink#1385] [Q&A (resolved)] [answered]
Hello,

I need access to the option bytes for an STM32F302 and as far as I understand it has been added to V1.8.0. I downloaded the code and compiled and installed it (worked flawlessly, great, thank you very much to all contributors!).

But when I run st-flash (or st-util) I get an error message: "/usr/local/bin/st-flash: symbol lookup error: /usr/local/bin/st-flash: undefined symbol: init_chipids". 

I searched for the problem and I have the feeling that it is related to libstlink being an older version but I could not find any solution. I am running Debian 11. Can anybody point me in the right direction how to solve the problem and getting 1.8.0 working?

Kind Regards,
Hugo

**Comment by Nightwalker-87** [comment, marked as answer]
@Hugo-Habicht The answer to this topic can be found in issue #1376.
Please either upgrade to Debian 12 to use `v1.8.0` or stay with `v1.7.0` on the current system.

**Comment by Hugo-Habicht** [comment]
Hello Nightwalker,

thank you very much for explaining where the problem is. Had not found #1376 myself.
I am afraid upgrading the OS is no option for me and v1.7.0 does not have option byte programming.

But I solved the problem with a few lines of code in my software simply checking the option byte value and then program it to memory read protection. This runs only the first time. Only obstacle was that, opposed to what the STM data sheet says, a power on reset is required in any case for the change to take effect. Was a bit of a problem in my battery operated system.

Kind regards,
